src/main/java/com/notably/logic/suggestion/SuggestionEngine.java

```diff
@@ -0,0 +1,14 @@
+package com.notably.logic.suggestion;
+
+/**
+ * Represents the Suggestion Engine instance.
+ * This is the class that we interact with when the user is typing an input.
+ */
+interface SuggestionEngine {
+    /**
+     * Generates suggestions.
+     * When input cannot get corrected, the "error" message will be shown in commandTextProperty.
```

Comment:
Nit: If you're writing about how error gets displayed in `commandTextPropoperty`, why not also do so for the list of suggestions (like where they are going to be added to)?